railties/test/generators/scaffold_generator_test.rb

```diff
@@ -383,8 +383,10 @@ def test_scaffold_generator_ignores_commented_routes
     assert_file "config/routes.rb", /\.routes\.draw do\n  resources :products\n/
   end
 
-  def test_scaffold_generator_with_switch_resource_route_false
-    run_generator [ "posts", "--resource-route=false" ]
+  def test_scaffold_generator_with_switch_no_resource_route
+    output = run_generator [ "posts", "--no-resource-route" ]
+
+    assert_no_match(/error/, output)
```

Comment:
When the --resource-route=false option is used, an error occurs because, within ScaffoldControllerGenerator, resource_route is defined as required rather than as a boolean. In Thor, specifying required for a class_option implies that the expected value should be a string. Consequently, setting resource_route to false is interpreted as a string, leading to an error being raised.

```
% bin/rails g scaffold Post name:string --resource-route=false
      invoke  active_record
      create    db/migrate/20240306105954_create_posts.rb
      create    app/models/post.rb
      invoke    test_unit
      create      test/models/post_test.rb
      create      test/fixtures/posts.yml
      invoke  scaffold_controller
      create    app/controllers/posts_controller.rb
      invoke    erb
      create      app/views/posts
      create      app/views/posts/index.html.erb
      create      app/views/posts/edit.html.erb
      create      app/views/posts/show.html.erb
      create      app/views/posts/new.html.erb
      create      app/views/posts/_form.html.erb
      create      app/views/posts/_post.html.erb
       error    false [not found]
      invoke    test_unit
      create      test/controllers/posts_controller_test.rb
      create      test/system/posts_test.rb
      invoke    helper
      create      app/helpers/posts_helper.rb
      invoke      test_unit
      invoke    jbuilder
      create      app/views/posts/index.json.jbuilder
      create      app/views/posts/show.json.jbuilder
      create      app/views/posts/_post.json.jbuilder
```

So I changed --resource-route=false to --no-resource-route.
> All arguments can be set to nil (except required arguments), by supplying a no- or skip-variant.
https://github.com/rails/thor/wiki/Method-Options